src/containers/List/List.js

```diff
@@ -1,66 +1,39 @@
-import React, { useState, useEffect } from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useEffect } from 'react';
 import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 
 import ListItem from '../../components/ListItem';
-import { fetchGames } from '../../actions/gamesActions';
-import { fetchPlatforms } from '../../actions/platformsActions';
 import { changeTitleAction } from '../../actions/titleActions';
 import { useSortedList } from '../../hooks/useSortedList';
+import { fetchData } from '../../actions/categoryActions';
+
 import classes from './List.module.scss';
-import * as PropTypes from 'prop-types';
 
 const List = ({
-                games,
-                platforms,
-                category,
-                fetchGames,
-                fetchPlatforms,
+                categoryTitle,
+                categoryData,
+                fetchData,
                 changeTitleAction
               }) => {
-  const [categoryList, setCategoryList] = useState([]);
-  const { sortedList } = useSortedList(categoryList);
-
-  const getCategoryData = () => {
-    switch (category) {
-      case 'games':
-        return games ? games.results : [];
-      case 'platforms':
-        return platforms ? platforms.results : [];
-      default:
-        return [];
-    }
-  };
-
-  const getTitle = () => {
-    switch (category) {
-      case 'games':
-        return 'All games';
-      case 'platforms':
-        return 'All platforms';
-      default:
-        return 'No data';
-    }
-  };
+  const { sortedList } = useSortedList();
+  const getTitle = () => categoryData && (categoryData.seo_title || 'All platforms');
 
   useEffect(() => {
-    if (category === 'platforms' && !platforms) {
-      fetchPlatforms();
+    if (!categoryData) {
+      fetchData();
     }
-
-    if (category) {
-      setCategoryList(getCategoryData());
-      changeTitleAction(getTitle());
-    }
-  }, [category]);
+  }, []);
 
   useEffect(() => {
-    if (!games) {
-      fetchGames();
-    }
-  }, []);
+    changeTitleAction(getTitle());
+  }, [categoryTitle]);
 
-  const renderList = () => (sortedList || categoryList)
-    .map(item => <ListItem category={category} key={item.id} item={item}/>);
+  const renderList = () => {
+    if (!categoryData) return <div>No data</div>;
```

Comment:
No data stub should appear if you have data but it is empty
When there is no data at all you should show loader/skeleton